lib/diagnostic.rb

```diff
@@ -44,5 +51,12 @@ def transform(words)
   # space and return the result.
   def read_file(filename)
     # Your code goes here
+  File.open('filename') do |file|
+  str = ""
+  file.each do |line|
+  str.concat(line)
```

Comment:
Close. This way you'd have to `.chomp` the line and add ' ' . 

`line.chomp +  str` in this case.